hs_collection_resource/views.py

```diff
@@ -104,3 +111,151 @@ def update_collection_for_deleted_resources(request, shortkey, *args, **kwargs):
+    :param collection_res_obj: collection obj
+    :return: a list of coverage metadata dict
+    """
+    res_id = collection_res_obj.short_id
+    new_coverage_list = []
+    try:
+        lon_list = []
+        lat_list = []
+        time_list = []
+        output_spatial_projection_str = "WGS84 EPSG:4326"
+        output_spatial_units_str = "Decimal degrees"
+        for contained_res_obj in collection_res_obj.resources.all():
+            for cvg in contained_res_obj.metadata.coverages.all():
+                if cvg.type.lower() == "box":
+                    lon_list.append(float(cvg.value["eastlimit"]))
+                    lon_list.append(float(cvg.value["westlimit"]))
+                    lat_list.append(float(cvg.value["northlimit"]))
+                    lat_list.append(float(cvg.value["southlimit"]))
+                elif cvg.type.lower() == "point":
+                    lon_list.append(float(cvg.value["east"]))
+                    lat_list.append(float(cvg.value["north"]))
+                elif cvg.type.lower() == "period":
+                    try:
+                        if cvg.value.get("start", None) is not None:
+                            start_date = parser.parse(cvg.value["start"])
+                            time_list.append(start_date)
+                        if cvg.value.get("end", None) is not None:
+                            end_date = parser.parse(cvg.value["end"])
+                            time_list.append(end_date)
+                    except ValueError as ex:
+                        # skip the res if it has invalid datetime string
+                        logger.warning("_calculate_collection_coverages: Ignore unknown datetime string. Collection resource ID: {0}. "
+                                       "Contained res ID: {1}"
+                                       "Msg: {2} ".format(res_id, contained_res_obj.short_id, ex.message))
+
+        # spatial coverage
+        if len(lon_list) > 0 and len(lat_list) > 0:
+            value_dict = {}
+            type_str = 'point'
+            lon_min = min(lon_list)
+            lon_max = max(lon_list)
+            lat_min = min(lat_list)
+            lat_max = max(lat_list)
+            if lon_min == lon_max and lat_min == lat_max:
+                type_str = 'point'
+                value_dict['east'] = lon_min
+                value_dict['north'] = lat_min
+                value_dict['units'] = output_spatial_units_str
+            else:
+                type_str = 'box'
+                value_dict['eastlimit'] = lon_max
+                value_dict['westlimit'] = lon_min
+                value_dict['northlimit'] = lat_max
+                value_dict['southlimit'] = lat_min
+                value_dict['units'] = output_spatial_units_str,
+                value_dict['projection'] = output_spatial_projection_str
+
+            new_coverage_list.append({'type': type_str, 'value': value_dict, 'element_id_str': "-1"})
```

Comment:
@pkdash 
When a new coverage metadata obj gets created ob backend, we need inform frontend its element id to set correct "form action url" as users may change its values on frontend later (resulting a post request against the "form action url").